libkbfs/tlf_journal.go

```diff
@@ -814,6 +817,47 @@ func (j *tlfJournal) convertMDsToBranchIfOverThreshold(ctx context.Context) (
 	return true, nil
 }
 
+func (j *tlfJournal) doOnMDFlush(ctx context.Context,
+	rmds *RootMetadataSigned) error {
+	if j.onMDFlush != nil {
+		j.onMDFlush.onMDFlush(rmds.MD.TlfID(), rmds.MD.BID(),
+			rmds.MD.RevisionNumber())
+	}
+
+	// Remove saved blocks in chunks to avoid starving foreground file
```

Comment:
I think that's somewhat orthogonal for this PR, but the next PR #579 it actually depends on that property to keep all blocks around until the next flush, even for non-ResolveBranch cases where saved blocks naturally build up over time.